src/intervals/interval_operations/set_operations.jl

```diff
@@ -201,3 +201,17 @@ function _interiordiff(x::Interval, y::Interval)
     t = isguaranteed(x) & isguaranteed(y)
     return (_unsafe_interval(h₁, d, t), _unsafe_interval(h₂, d, t), _unsafe_interval(inter, d, t))
 end
+
+function interval_diff(x::Interval{T}, y::Interval) where {T<:NumTypes}
```

Comment:
I think this function should not be exported. It is a bit misleading as it is **not** the set difference of intervals, but some other sort of difference operation.

Reply:
The other option would be to document clearly the function since its name is ambiguous.